### PR TITLE
Fix flaky tests in SeekableStreamSupervisorStateTest

### PR DESCRIPTION
**Pull request by abhishekagarwal87**
This PR attempts to fix a few flaky tests and add more information to the assertion failure message

- SeekableStreamSupervisorStateTest - Tests in this class rely on a latch to know when a metric has been emitted. However, the latch is counted down for all the emitted metrics and not just the metric we want to test the emission of. This is why tests wait for the latch to be counted down twice. Except in one place, probably by accident, and that was a flaky test. Now with this change, we wait for latch to be released only for the metric being tested. I am also using the same latch mechanism to track emission of `notice time` metric (that was another flaky test)
- ITTLS test - We now retry requests for ClosedChannelException errors as well. This *may* fix the flakiness. 
- Added more information to the assertion failure message in IndexTaskTest and CompactionTaskParallelRunTest. 
- AbstractQueryResourceTestClient - Added retries for running queries. 

This PR has:
- [ ] been self-reviewed.

### CODE REVIEW

**Comment by paul-rogers** [comment]
@abhishekagarwal87, thanks for the fix! Really excited to see another source of error removed.

The changes look straight-forward enough. However, it is not clear to this newbie what they're trying to fix. Something about a latch? Can you add a note to explain the underlying problem a bit more?

**Comment by abhishekagarwal87** [comment]
@paul-rogers - I have added more explanation now in the PR. can you review again?

**Review by paul-rogers — COMMENTED** [review]
@abhishekagarwal87, nicely done, thanks!
LGTM (non-binding)

File: indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java [hunk 1]
```diff
@@ -1326,6 +1339,9 @@ protected Map<String, Long> getPartitionTimeLag()
     @Override
     protected void emitLag()
     {
+      if ((metricFlag & LAG) == 0) {
```
Comment (kfaraz) [inline indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java:1342]: This is a nice approach but we would only be able to track 8 metrics at most. Wouldn't it be simpler to just do `metricFlag != LAG`?

File: integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java [hunk 2]
```diff
@@ -537,4 +534,19 @@ private StatusResponseHolder makeRequest(
       throw new RuntimeException(e);
     }
   }
+
+  private boolean isRetriable(Throwable ex)
```
Comment (kfaraz) [inline integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java:538]: `isRetriable` seems a little off 😅. Maybe use `canRetry` or `shouldRetry`.

**Review by kfaraz — APPROVED** [review]
Overall LGTM. Added minor suggestions.

File: integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java (hunk 2 above)
Comment (abhishekagarwal87) [reply integration-tests/src/test/java/org/apache/druid/tests/security/ITTLSTest.java:538]: seems alright to me 😄

File: indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java (hunk 1 above)
Comment (abhishekagarwal87) [reply indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java:1342]: did it like this for scenarios when we want to test the emission of two different metrics. I don't have such a test case now so I am happy to change the implementation.

Comment (kfaraz) [reply indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java:1342]: No, that makes sense. As it is just inside this test, this is fine too. We can change it later if we need to.

**Pull request merged**